Homeworks/Homework_3/script.py

```diff
@@ -0,0 +1,263 @@
+import argparse as argp
+import csv
+import requests as re
+import logging
+import os
+from datetime import datetime
+from pytz import timezone
+from timezonefinder import TimezoneFinder
+from geopy.geocoders import Nominatim
+from collections import Counter
+import shutil
+import zipfile
+
+
+def download_csv_file():
+    """
+    Download data from web randomuser using requests lib
+    :return: string-type data in csv format
+    """
+    url = 'https://randomuser.me/api/?results=50&format=csv'
+    response = re.get(url)
+    return response.text
+
+
+def write_data_to_csv(csv_data, csv_file_path):
+    """
+    Write data to csv file using csv lib
+    :param csv_data: str, data that was downloaded and converted into csv format
+    :param csv_file_path: str, path to the folder where the csv file is located
+    """
+    with open(csv_file_path, 'w', encoding='utf-8') as csv_file:
+        csv_file.write(csv_data)
+
+
+def read_data_from_csv_file(csv_file_path):
+    """
+    Read data from csv file and write it to variable using csv lib
+    :param csv_file_path: str, path to the folder where the csv file is located
+    :return: list, data that was read from the csv file
+    """
+    dict_data = []
+    with open(csv_file_path, 'r', encoding='utf-8') as csv_file:
+        csv_reader = csv.DictReader(csv_file)
+        for row in csv_reader:
+            dict_data.append(row)
+    return dict_data
+
+
+def filter_data(csv_data, filtering):
+    """
+    Filter data using match cases
+    :param csv_data: str, data which has to be filtered
+    :param filtering: list, list with information about filtering
+                      where the first value is a type of filtering: gender or number;
+                      the second value is a gender by which the data has to be filtered;
+                      the third value is ф number of rows to be filtered
+    :return: list, filtered data
+    """
+    match filtering[0]:
+        case 'gender':
+            return list(filter(lambda row: row['gender'] == filtering[1], csv_data))
+        case 'number':
+            return csv_data[:filtering[2]]
+        case _:
+            return csv_data
+
+
+def add_new_fields(csv_data):
+    """
+    Add new fields and change others in the data using match
+    cases and timezonefinder, geopy.geocoders, pytz, datetime libs
+    :param csv_data: list, data to which new fields will be added and others will be changed
+    :return: list, data with added and changed values
+    """
+    for i, user_info in enumerate(csv_data):
+        # ADDING FIELD 'global_index'
+        user_info['global_index'] = i + 1
```

Comment:
https://www.geeksforgeeks.org/enumerate-in-python/

Reply:
Could you explain this issue? I don`t understand what's wrong :(